crates/bevy_ecs/src/reflect/bundle.rs

```diff
@@ -35,7 +35,8 @@ pub struct ReflectBundleFns {
     /// Function pointer implementing [`ReflectBundle::apply_or_insert()`].
     pub apply_or_insert: fn(&mut EntityWorldMut, &dyn PartialReflect, &TypeRegistry),
     /// Function pointer implementing [`ReflectBundle::remove()`].
-    pub remove: fn(&mut EntityWorldMut),
+    pub remove: fn(&mut EntityWorldMut) -> ReflectBundle,
+    pub take: fn(&mut EntityWorldMut) -> Option<ReflectBundle>,
```

Comment:
I think one downside to this approach is that it couples these pointers to `ReflectBundle`, making them harder to manually construct outside of a `ReflectBundle` (which iirc is part of the reason they were separated into their own struct)